File: Scripts/Daily_count.py
from Functions import *
import pandas as pd
import os


def create_daily_dataframe(index: list):
    data = pd.DataFrame(index=index,
                        columns=["Count"])
    data.index.names = ["Date"]
    data = data.fillna(0)
    return data


parameters = {"path data": "../Data/",
              "path output": "../Output/",
              "useless columns": ["Usuario_Id",
                                  "Genero",
                                  "Año_de_nacimiento",
                                  "Inicio_del_viaje",
                                  "Fin_del_viaje",
                                  "Origen_Id",
                                  "Destino_Id"]}
files = sorted(os.listdir(parameters["path data"]))
period = obtain_period_from_filenames(files)
dates = obtain_consecutive_dates_from_period(period)
daily_count = create_daily_dataframe(dates)
for file in files:
    print("Analizando archivo {}".format(file))
    data = read_data(parameters["path data"],
                     file)
    data.index = pd.to_datetime(data["Inicio_del_viaje"])
    data = data.drop(columns=parameters["useless columns"])
    data = data.resample("D").count()
    for index in data.index:
        date = (data["Viaje_Id"][index])
        daily_count["Count"][index.date()] = date
daily_count.to_csv("{}Daily_count.csv".format(parameters["path output"]))

File: Scripts/dailyCount.R
dir <- getSrcDirectory(function(x) {x})
wdir <- sub("Scripts", "Data", dir)

setwd(wdir)

df = data.frame(Date=c(), Count=c())
for( datafile in list.files()) {
     counts <- c()
    data <- read.csv(datafile, header = TRUE)
    
    dates <- as.Date(unique(gsub("( ).*", "", unlist(data["Inicio_del_viaje"]))))
    data_list <- as.Date(gsub("( ).*", "",unlist(data["Inicio_del_viaje"])))
    
    for(i in 1:length(dates)) {
      counts[i] <- length(data_list[data_list==dates[i]])
    }
    if(!exists("df")) {
      df <- data.frame(Date=dates, Count = counts)
    } else {
      tmp <- data.frame(Date=dates, Count = counts)
      df <- rbind(df, tmp)
      rm(tmp)
    }
    rm(data)
    rm(dates)
    rm(data_list)
}
write.table(df, file="../Output/Daily_count.csv", sep=",", row.names = FALSE, quote = FALSE)